vlan dot1q support

prashanthjbabu opened the pull request:
This commit adds DOT1Q(VLAN) support so that JOY maybe able to parse such packets! This handles Issue -  "802.1Q Virtual LAN #77"

fliphil commented:
Could you please supply us with the PCAP file that you used to test? If it's not private you can throw it into the "resources" directory. Otherwise, you could email it directly.

fliphil commented on src/pkt_proc.c:
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
+           default :
+               return;
+           }
+       break;
```
please indent this break 1 time to the right.

fliphil commented on src/pkt_proc.c:
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
+       case ETH_TYPE_IP:
+           ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN);
+           ip_hdr_len = ip_hdr_length(ip);
+       break;
```
please indent this break 1 time to the right.

fliphil commented on src/pkt_proc.c:
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
+                   ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN + DOT1Q_HDR_LEN);
+                   ip_hdr_len = ip_hdr_length(ip);
+                   break;
+           default :
```
please indent this default block 1 time to the right.

fliphil commented on src/pkt_proc.c:
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
-    /* define/compute ip header offset */
-    ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN);
-    ip_hdr_len = ip_hdr_length(ip);
+    ether_type = ntohs(*(uint16_t *)(packet + 12));//Offset to get ETH_TYPE
```
A comment here explaining the purpose of this code block, i.e. "Support of both normal ethernet and 802.1q. Distinguish between the two accepted types."

prashanthjbabu replied:
Thanks! Comments addressed in new commit !

fliphil commented on src/pkt_proc.c:
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
+           ip_hdr_len = ip_hdr_length(ip);
+       break;
+       case ETH_TYPE_DOT1Q:
+           //Offset to get VLAN_TYPE
```
please place joy_log_info("Ethernet type - 802.1q vlan")

fliphil commented on src/pkt_proc.c:
```diff
@@ -888,18 +889,36 @@ void process_packet (unsigned char *ignore, const struct pcap_pkthdr *header,
-    ip_hdr_len = ip_hdr_length(ip);
+    ether_type = ntohs(*(uint16_t *)(packet + 12));//Offset to get ETH_TYPE
+    switch(ether_type) {
+       case ETH_TYPE_IP:
```
place joy_log_info("Ethernet type - normal")

prashanthjbabu commented:
Thanks Philip! i've addressed your comments in a new commit!

On Wed, Aug 9, 2017 at 12:38 AM, Philip Perricone <notifications@github.com>
wrote:

> *@fliphil* commented on this pull request.
> ------------------------------
>
> In src/pkt_proc.c
> <https://github.com/cisco/joy/pull/83#discussion_r132004120>:
>
> >
>      flocap_stats_incr_num_packets();
>      fprintf(info, "\n");
>      joy_log_info("++++++++++ Packet %lu ++++++++++", flocap_stats_get_num_packets());
>      //  packet_count++;
>
>      // ethernet = (struct ethernet_hdr*)(packet);
> -
> -    /* define/compute ip header offset */
> -    ip = (struct ip_hdr*)(packet + ETHERNET_HDR_LEN);
> -    ip_hdr_len = ip_hdr_length(ip);
> +    ether_type = ntohs(*(uint16_t *)(packet + 12));//Offset to get ETH_TYPE
> +    switch(ether_type) {
> +       case ETH_TYPE_IP:
>
> place joy_log_info("Ethernet type - normal")
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cisco/joy/pull/83#pullrequestreview-55039693>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AB2Fwz_if3LQwnUlWNZma_TkARwEmjaSks5sWLIZgaJpZM4Ownuo>
> .
>

fliphil commented:
#83
